### PR TITLE
add support for postgresql to the installer

### PR DESCRIPTION
**Pull request by jovestefanovski**
WIP: Implementation of postgresql in the installer.

### CODE REVIEW

File: vagrant/provisioning/arkcase-ce.yml [hunk 1]
```diff
@@ -13,6 +13,8 @@
       tags: [core, ldap]
     - role: mariadb
       tags: [core, mariadb]
+    - role: postgresql
+      tags: [core, postgresql]
```
Comment (bojmil-armedia) [inline vagrant/provisioning/arkcase-ce.yml:25]: I wouldn't add this one as core role. By default we want mariadb.

File: vagrant/provisioning/arkcase-ce.yml [hunk 2]
```diff
@@ -27,6 +31,12 @@
       tags: [core, alfresco]
     - role: alfresco-site
       tags: [core, alfresco, alfresco-site]
+    - role: alfresco7-setup
+      tags: [alfresco7, alfresco7-setup]
+    - role: alfresco7-ce
+      tags: [alfresco7, alfresco7-ce]
+    - role: alfresco7
+      tags: [alfresco7]
```
Comment (dzdimov) [inline vagrant/provisioning/arkcase-ce.yml]: should these roles be before alfresco-site or after?

**Comment by jovestefanovski** [comment]
Do not merge in develop.
Please review this code.

File: vagrant/provisioning/roles/arkcase-app/tasks/main.yml [hunk 3]
```diff
@@ -245,7 +245,7 @@
 
 - name: Ensure ArkCase required services are populated inside the arkcase.service
   set_fact:
-    arkcase_required_services: "{{ (db_engine == 'postgresql') | ternary('postgresql-13.service', 'mariadb.service') }} {{'activemq.service alfresco.service samba.service solr.service config-server.service' if ldap_type=='samba' else 'activemq.service alfresco.service solr.service config-server.service' }}"
+    arkcase_required_services: "{{ (db_engine == 'postgresql') | ternary('postgresql-13.service alfresco7.service', 'mariadb.service alfresco.service') }} {{'activemq.service samba.service solr.service config-server.service' if ldap_type=='samba' else 'activemq.service solr.service config-server.service' }}"
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/arkcase-app/tasks/main.yml]: we need a better solution here, to support multi-VM installs where every service is not on the same VM.  Let's try to think this through.

File: vagrant/provisioning/roles/alfresco-change-password/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  set_fact:
+    alfresco_url: "{{ alfresco_protocol }}://{{ alfresco_host }}:{{ alfresco_port }}"
+
+- name: change alfresco admin password
+  uri:
+    url: "{{ alfresco_url }}/alfresco/s/api/person/changepassword/admin"
+    method: POST
+    user: admin
+    password: "{{ old_alfresco_admin_password }}"
+    validate_certs: no
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/alfresco-change-password/tasks/main.yml:21]: @jovestefanovski , @dzdimov , this is my new approach for setting the Alfresco password.  you would call this role after alfresco, and before alfresco_site, and you would specify the old_alfresco_admin_password as an additional extra fact.

File: vagrant/provisioning/roles/alfresco/files/passencode.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env python3
+
+import bcrypt
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/alfresco/files/passencode.py:3]: with using the web script, we don't need the bcrypt changes any more, and we don't need the database update. The database update didn't work for me anyway, at least not on PostgreSQL.

File: vagrant/provisioning/roles/alfresco/tasks/main.yml [hunk 6]
```diff
@@ -277,6 +287,7 @@
       synchronization.syncWhenMissingPeopleLogIn=true
       synchronization.autoCreatePeopleOnLogin=false
       synchronization.import.cron=0 0/5 * * * ? *
+      system.preferred.password.encoding=bcrypt10
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/alfresco/tasks/main.yml:333]: Now this setting change definitely does make sense.

**Comment by david-oc-miller** [comment]
@bojmil-armedia , @jovestefanovski , @dzdimov , this is ready to merge.  PostgreSQL on RHEL8 works, and PostgreSQL on CentOS 7 works, and MariaDB on RHEL8 works, and MariaDB on CentOS7 works.

All that is left is to copy the arkcase-ce-facts.yml and arkcase-ce.yml updates, to e.g. arkcase-dev-facts.yml and arkcase-dev.yml and whatnot.

I'll merge this in next few days, unless I see code review comments from one of you.

File: vagrant/provisioning/arkcase-ce.yml (hunk 1 above)
Comment (david-oc-miller) [reply vagrant/provisioning/arkcase-ce.yml:25]: The `when` clause skips this role when the `db_engine` is not `postgresql`.  I have tested this; it works well.

**Comment by bojmil-armedia** [comment]
This one is a pretty big PR. Looks good.

**Pull request merged**